src/nvim/screen.c

```diff
@@ -6153,6 +6153,10 @@ void clearmode(void)
 
 static void recording_mode(int attr)
 {
+  if (p_ch <= 0) {
+    return;
```

Comment:
Here's another case where we're adding guards just to avoid a message. Maybe we should change the message code now to _never show hit-enter_ when `p_ch==0`.

Reply:
OK.